wikipedia/wikipedia_scratcher.py

```diff
@@ -95,11 +93,20 @@ def get_response_elems(language="en"):
                 file=sys.stderr,
             )
             return {}
-        elif "query" not in search_data:
-            print(f"search data is: \n{search_data}", file=sys.stderr)
-            sys.exit(1)
-        else:
-            raise e
+
+        search_data_dict = search_data["query"]["statistics"]
+        search_data_dict["language"] = language
+        return search_data_dict
+
+    except requests.HTTPError as e:
+        print(f"HTTP Error: {e}", file=sys.stderr)
+        return {}
+    except requests.RequestException as e:
+        print(f"Request Exception: {e}", file=sys.stderr)
+        return {}
+    except KeyError as e:
+        print(f"KeyError: {e}. Search data is: {search_data}", file=sys.stderr)
+        return {}
```

Comment:
made the below change
except requests.HTTPError as e:
        print(f"HTTP Error: {e}", file=sys.stderr)
        raise
    except requests.RequestException as e:
        print(f"Request Exception: {e}", file=sys.stderr)
        raise
    except KeyError as e:
        print(f"KeyError: {e}. Search data is: {search_data}", file=sys.stderr)
        raise
Instead of printing messages and continuing execution, errors are raised using raise. @TimidRobot  please check it once